Bump upstream v1.12.8

alexpeterson91 opened the pull request:
Bump upstream to v1.12.8

alexpeterson91 committed:
Bump upstream

bump upstream

alexpeterson91 committed:
Bump Upstream v1.12.8

bump Upstream

alexpeterson91 commented:
Build still fails this repo is confusing it seems it pulls from multiple upstream repos and following the upstream repos they link back and forth with older repos that have been archived then back. The workflow shows it's pulling from the new repo initially. But further down it pulls from the old. I haven't dug into it really yet since I feel like I'm one of the only people running this package and for me it's purely for testing and altruistic purposes.

q9f commented:
pinging @meowsbits - do you know why this build fails? the logs are not really helpful.

```
#12 [builder 3/3] RUN git clone -b v1.12.8 https://github.com/etclabscore/core-geth.git &&     cd core-geth && make geth
#0 0.063 Cloning into 'core-geth'...
#12 13.31 Note: switching to '654bc751dffdecb3670cb3a1d6f1915fb774fc58'.
#12 13.31 
#12 13.31 You are in 'detached HEAD' state. You can look around, make experimental
#12 13.31 changes and commit them, and you can discard any commits you make in this
#12 13.31 state without impacting any branches by switching back to a branch.
#12 13.31 
#12 13.31 If you want to create a new branch to retain commits you create, you may
#12 13.31 do so (now or later) by using -c with the switch command. Example:
#12 13.31 
#12 13.31   git switch -c <new-branch-name>
#12 13.31 
#12 13.31 Or undo this operation with:
#12 13.31 
#12 13.31   git switch -
#12 13.31 
#12 13.31 Turn off this advice by setting config variable advice.detachedHead to false
#12 13.31 
#12 13.81 env GO111MODULE=on go run build/ci.go install ./cmd/geth
#12 14.58 go: downloading golang.org/x/crypto v0.0.0-20210921155107-089bfa567519
#12 14.58 go: downloading github.com/cespare/cp v0.1.0
#12 14.59 go: downloading github.com/Azure/azure-sdk-for-go/sdk/storage/azblob v0.3.0
#12 14.59 go: downloading github.com/tidwall/gjson v1.9.3
#12 14.60 go: downloading github.com/davecgh/go-spew v1.1.1
#12 14.63 go: downloading github.com/tidwall/pretty v1.2.0
#12 14.63 go: downloading github.com/tidwall/match v1.1.1
#12 14.78 go: downloading github.com/Azure/azure-sdk-for-go/sdk/azcore v0.21.1
#12 14.78 go: downloading github.com/Azure/azure-sdk-for-go/sdk/internal v0.8.3
#12 15.01 package command-line-arguments
#12 15.01 	imports github.com/ethereum/go-ethereum/internal/build
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/storage/azblob
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/azcore: build constraints exclude all Go files in /go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v0.21.1
#12 15.01 package command-line-arguments
#12 15.01 	imports github.com/ethereum/go-ethereum/internal/build
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/storage/azblob
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/shared: build constraints exclude all Go files in /go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v0.21.1/internal/shared
#12 15.01 package command-line-arguments
#12 15.01 	imports github.com/ethereum/go-ethereum/internal/build
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/storage/azblob
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/azcore/policy: build constraints exclude all Go files in /go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v0.21.1/policy
#12 15.01 package command-line-arguments
#12 15.01 	imports github.com/ethereum/go-ethereum/internal/build
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/storage/azblob
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/azcore/streaming: build constraints exclude all Go files in /go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v0.21.1/streaming
#12 15.01 package command-line-arguments
#12 15.01 	imports github.com/ethereum/go-ethereum/internal/build
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/storage/azblob
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/azcore/to: build constraints exclude all Go files in /go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v0.21.1/to
#12 15.01 package command-line-arguments
#12 15.01 	imports github.com/ethereum/go-ethereum/internal/build
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/storage/azblob
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/internal/uuid: build constraints exclude all Go files in /go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/internal@v0.8.3/uuid
#12 15.01 package command-line-arguments
#12 15.01 	imports github.com/ethereum/go-ethereum/internal/build
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/storage/azblob
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/internal: build constraints exclude all Go files in /go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/storage/azblob@v0.3.0/internal
#12 15.02 make: *** [Makefile:13: geth] Error 1
#12 ERROR: process "/bin/sh -c git clone -b ${UPSTREAM_VERSION} https://github.com/etclabscore/core-geth.git &&     cd core-geth && make geth" did not complete successfully: exit code: 2
------
 > [builder 3/3] RUN git clone -b v1.12.8 https://github.com/etclabscore/core-geth.git &&     cd core-geth && make geth:
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/azcore/to: build constraints exclude all Go files in /go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v0.21.1/to
#12 15.01 package command-line-arguments
#12 15.01 	imports github.com/ethereum/go-ethereum/internal/build
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/storage/azblob
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/internal/uuid: build constraints exclude all Go files in /go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/internal@v0.8.3/uuid
#12 15.01 package command-line-arguments
#12 15.01 	imports github.com/ethereum/go-ethereum/internal/build
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/storage/azblob
#12 15.01 	imports github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/internal: build constraints exclude all Go files in /go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/storage/azblob@v0.3.0/internal
#12 15.02 make: *** [Makefile:13: geth] Error 1
------
Dockerfile:8
--------------------
   7 |     
   8 | >>> RUN git clone -b ${UPSTREAM_VERSION} https://github.com/etclabscore/core-geth.git && \
   9 | >>>     cd core-geth && make geth
  10 |     
--------------------
error: failed to solve: process "/bin/sh -c git clone -b ${UPSTREAM_VERSION} https://github.com/etclabscore/core-geth.git &&     cd core-geth && make geth" did not complete successfully: exit code: 2
```

meowsbits commented:
The Go version is too low. Try Go v1.17 or higher (current is v1.18.3). 
Upstream ethereum/go-ethereum will fail in the same way.

eg.

```
diff --git a/Dockerfile b/Dockerfile
index 70299190f..ce82ef0f0 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -4,7 +4,7 @@ ARG VERSION=""
 ARG BUILDNUM=""
 
 # Build Geth in a stock Go builder container
-FROM golang:1.18-alpine as builder
+FROM golang:1.15-alpine as builder
```
> https://github.com/etclabscore/core-geth/blob/master/Dockerfile#L7=

meowsbits committed:
Update Dockerfile: use Go v1.18

This resolves an issue encountered here https://github.com/dappnode/DAppNodePackage-ethereum-classic/pull/16

alexpeterson91 committed:
Merge pull request #17 from meowsbits/meowsbits-patch-1

Update Dockerfile: use Go v1.18

alexpeterson91 commented:
Testing now, this PR and Branch will likely be deleted and replaced by the CI workflow from #18.
this can be tested at 

```
DNP (DAppNode Package) built and uploaded 
  Release hash : /ipfs/QmbQiwZ6e4MJhGCF8MDZeZWTC6xwcgEPFJBckK5bgJy7fL
```
  http://my.dappnode/#/installer/%2Fipfs%2FQmbQiwZ6e4MJhGCF8MDZeZWTC6xwcgEPFJBckK5bgJy7fL

alexpeterson91 commented:
New CI  created PR and other PRs merged, tested, and ready for publishing

alexpeterson91 commented:
New CI created PR #18 and other PRs merged, tested, and ready for publishing have made this PR no longer needed
